usage/rest-client/src/main/java/org/apache/brooklyn/rest/client/BrooklynApi.java

```diff
@@ -114,11 +150,23 @@ public BrooklynApi(String endpoint, @Nullable Credentials credentials) {
         }
     }
 
+    /**
+     * @deprecated since 0.9.0. Use {@link BrooklynApiFactory#getBrooklynApi(java.net.URL, String, String)} instead
+     * @see org.apache.brooklyn.rest.client.DefaultBrooklynApiFactory
+     * @see org.apache.brooklyn.rest.client.PoolingBrooklynApiFactory
+     */
+    @Deprecated
```

Comment:
Yup, it's a little too protective there. What if we change to `protected` and let the user subclass the `BrooklynApi` to provide an alternative implementation?